src/fides/api/models/privacy_experience.py

```diff
@@ -1,94 +1,252 @@
-from typing import Any, Dict, List, Optional, Set, Tuple, Type
+from typing import Any, Dict, List, Optional, Set, Type
+from uuid import uuid4
 
 from sqlalchemy import Boolean, Column
 from sqlalchemy import Enum as EnumColumn
-from sqlalchemy import Float, ForeignKey, String, UniqueConstraint, and_, or_
-from sqlalchemy.orm import Query, Session, relationship
-from sqlalchemy.util import hybridproperty
+from sqlalchemy import Float, ForeignKey, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import ARRAY, JSONB
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.orm import Query, RelationshipProperty, Session, relationship
+from sqlalchemy.orm.dynamic import AppenderQuery
 
 from fides.api.db.base_class import Base
+from fides.api.models.custom_asset import CustomAsset
 from fides.api.models.privacy_notice import (
-    ConsentMechanism,
     PrivacyNotice,
     PrivacyNoticeRegion,
     create_historical_data_from_record,
     update_if_modified,
 )
+from fides.api.schemas.language import SupportedLanguage
 
-BANNER_CONSENT_MECHANISMS: Set[ConsentMechanism] = {
-    ConsentMechanism.notice_only,
-    ConsentMechanism.opt_in,
-}
+
+class ExperienceNotices(Base):
+    """
+    A many-to-many table that linkes Privacy Notices to shared Privacy Experience Configs.
+    """
+
+    def generate_uuid(self) -> str:
+        """
+        Generates a uuid with a prefix based on the tablename to be used as the
+        record's ID value
+        """
+        try:
+            prefix = f"{self.current_column.table.name[:3]}_"  # type: ignore
+        except AttributeError:
+            prefix = ""
+        uuid = str(uuid4())
+        return f"{prefix}{uuid}"
+
+    # Overrides Base.id so this is not a primary key.
+    # Instead, we have a composite PK of notice_id and experience_config_id
+    id = Column(String(255), default=generate_uuid)
+
+    notice_id = Column(
+        String,
+        ForeignKey("privacynotice.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        primary_key=True,
+    )
+    experience_config_id = Column(
+        String,
+        ForeignKey("privacyexperienceconfig.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        primary_key=True,
+    )
 
 
 class ComponentType(Enum):
     """
     The component type - not formalized in the db
+
+    Overlay type has been deprecated but can't be removed for backwards compatibility
+    without significant data migrations.
     """
 
-    overlay = "overlay"
+    overlay = "overlay"  # Deprecated. DO NOT REMOVE.
+    banner_and_modal = "banner_and_modal"
+    modal = "modal"
     privacy_center = "privacy_center"
-    tcf_overlay = "tcf_overlay"
+    tcf_overlay = "tcf_overlay"  # TCF Banner + modal combined
+
+
+# Fides JS UX Types - there should only be one of these defined per region
+FidesJSUXTypes: List[ComponentType] = [
+    ComponentType.banner_and_modal,
+    ComponentType.modal,
+]
```

Comment:
Plus adds application behavior to prevent a region from being on more than one Fides JS UX Type at a time.  This was previously more of a database constraint.